app/controllers/developers_controller.rb

```diff
@@ -51,6 +52,15 @@ def require_new_developer!
     end
   end
 
+  def invisible?
+    @developer = Developer.find(params[:id])
+    if @developer.search_status == "invisible"
+      unless user_signed_in? && current_user == @developer.user
+        redirect_to root_path, notice: t(".invisible")
+      end
+    end
+  end
```

Comment:
This logic can be moved to `DeveloperPolicy`, which is powered by Pundit. Check out `MessagingPolicy` for an example on something very similar.